main.rb

```diff
@@ -0,0 +1,145 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/ModuleLength, Style/For
+module Enumerable
+  # 1.my_each
+  def my_each
+    for i in self
+      yield(i) if block_given?
+    end
+    self
+  end
+
+  # 2. my_each_with_index
+  def my_each_with_index
+    for i in (0..self).size - 1
+      yield(self[i], i) if block_given?
+    end
+    self
+  end
```

Comment:
- #my_each_with_index should return an enumerator if no block is given
- #my_each_with_index does not mutate the original array
- #my_each_with_index when a block is given when self is an array returns the array itself after calling the given block once for each element in self
- #my_each_with_index when a block is given when self is a range behaves like #each_with_index calls the given block once for each element in self
- #my_each_with_index when a block is given when self is a hash behaves like #each_with_index calls the given block once for each element in self
- #my_each_with_index when a block is given when self is an array behaves like #each_with_index calls the given block once for each element in self